x/airdrop/module.go

```diff
@@ -0,0 +1,142 @@
+func (a AppModuleBasic) Name() string {
+	return types.ModuleName
+}
+
+// RegisterLegacyAminoCodec does nothings. Airdrop does not support amino
+func (a AppModuleBasic) RegisterLegacyAminoCodec(amino *codec.LegacyAmino) {}
+
+// RegisterInterfaces registers module concrete types into protobuf Any.
+func (a AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	types.RegisterInterfaces(registry)
+}
+
+// DefaultGenesis returns default genesis state as raw bytes for the airdrop module
+func (a AppModuleBasic) DefaultGenesis(cdc codec.JSONMarshaler) json.RawMessage {
+	return cdc.MustMarshalJSON(types.DefaultGenesisState())
+}
+
+// ValidateGenesis performs genesis state validation for the airdrop module.
+func (a AppModuleBasic) ValidateGenesis(cdc codec.JSONMarshaler, _ client.TxEncodingConfig, bz json.RawMessage) error {
+	var gs types.GenesisState
+	if err := cdc.UnmarshalJSON(bz, &gs); err != nil {
+		return fmt.Errorf("failed to unmarshal %s genesis state: %w", host.ModuleName, err)
+	}
+
+	return gs.Validate()
+}
+
+// RegisterRESTRoutes does nothing. Airdrop does not support legacy REST routes.
+func (a AppModuleBasic) RegisterRESTRoutes(context client.Context, router *mux.Router) {}
+
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the ibc module.
+func (a AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
+	types.RegisterQueryHandlerClient(context.Background(), mux, types.NewQueryClient(clientCtx))
+}
+
+// GetTxCmd returns the root tx command for the airdrop module.
+func (a AppModuleBasic) GetTxCmd() *cobra.Command {
+	return cli.GetTxCmd()
+}
+
+// GetQueryCmd returns the root query command for the airdrop module.
+func (a AppModuleBasic) GetQueryCmd() *cobra.Command {
+	return cli.GetQueryCmd()
+}
+
+// AppModule implements an application module for the airdrop module.
+type AppModule struct {
+	AppModuleBasic
+
+	keeper *keeper.Keeper
+}
+
+// NewAppModule creates a new AppModule object
+func NewAppModule(k *keeper.Keeper) AppModule {
+	return AppModule{
+		keeper: k,
+	}
+}
+
+// Name returns the airdrop module's name.
+func (a AppModule) Name() string {
+	return types.ModuleName
+}
+
+// InitGenesis performs genesis initialization for the airdrop module
+func (a AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONMarshaler, data json.RawMessage) []abci.ValidatorUpdate {
+	var genesisState types.GenesisState
+	cdc.MustUnmarshalJSON(data, &genesisState)
+	a.keeper.InitGenesis(ctx, &genesisState)
+	return []abci.ValidatorUpdate{}
+}
+
+// ExportGenesis returns the exported genesis state as raw bytes for the airdrop module
+func (a AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONMarshaler) json.RawMessage {
+	gs := a.keeper.ExportGenesis(ctx)
+	return cdc.MustMarshalJSON(gs)
+}
+
+// RegisterInvariants registers the airdrop module invariants.
+func (a AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {
+	// TODO
```

Comment:
bump to not forget

Reply:
No invariant in this version of the module